lib/main/atom/views/outline/navigationNodeComponent.tsx

```diff
@@ -0,0 +1,114 @@
+import * as etch from "etch"
+import {isEqual} from "lodash"
+import {Props, NavigationTreeViewModel} from "./semanticViewModel"
+
+export class NavigationNodeComponent implements JSX.ElementClass {
+  constructor(public props: Props) {
+    etch.initialize(this)
+  }
+
+  public async update(props: Partial<Props>) {
+    this.props = {...this.props, ...props}
+    await etch.update(this)
+  }
+
+  public async destroy() {
+    this.props.root = null
+    await etch.destroy(this)
+  }
+
+  render() {
+    return this.renderNode(this.props.navTree)
+  }
+
+  private renderNode(node: NavigationTreeViewModel | null): JSX.Element {
+    if (node === null) return <div />
+
+    const _root = this.props.root
+    if (!_root) return <div />
+
+    const selected =
+      (_root.selectedNode && this.isSameNode(node, _root.selectedNode)) ||
+      (!_root.selectedNode && _root.isSelected(node))
```

Comment:
basically, `_root.isSelected()` is needed, if there's no node set as selected (yet):  
it calculates if the current cursor is within code-position that the node represents

this processing needs to be done, since it not enough to just check, if the cursor is within node's start- & end-range, as there could be a "more specfic" node within the start- & end-range that should be selected instead.

But I guess this is one the utility functions, that should get extracted